src/Components/BeerDetail.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component } from 'react';
+import axios from 'axios';
+import { Link } from 'react-router-dom';
+import Nav from './Nav';
+
+class BeerDetail extends Component {
+  state = {
+    oneBeer: [],
+  };
+
+  componentDidMount() {
+    this.getBeer();
+  }
+
+  getBeer = () => {
+    const { params } = this.props.match;
+    axios
+      .get(`https://api.punkapi.com/v2/beers/${params.id}`)
+      .then((responseFromApi) => {
+        this.setState({ oneBeer: responseFromApi.data });
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  };
+
+  render() {
+    return (
+      <div className = "beer-details"> 
+      <div className = "nav">
+        <Nav />
+        </div>
+        {this.state.oneBeer.map((beer, index) => {
+          return (
+            <div className="single-beer-details" key={index}>
```

Comment:
It's better practice to use an id for the `key` attribute when you have the option.
```
<div className="single-beer-details" key={beer._id}>